proteus/BoundaryConditions.pyx

```diff
@@ -47,7 +47,7 @@ class BC_Base(object):
             cls.ct = context
 
 
-class BoundaryCondition():
+class BoundaryCondition(BC_Base):
```

Comment:
@cekees `BoundaryCondition` is actually not a subclass of `BC_Base`. Maybe the names are a bit confusing, I made this class for cython and for initializing variables that are originally tied to an object but do not change in the `uOfXT(x,t)` function during the simulation. 
`BC_Base` is a container that could have class instances of `BoundaryCondition` as variables that would then be manipulated through functions (like for `BC_RANS` in mprans.BoundaryConditions where `u_advective`, `vof_dirichlet`, etc are `BoundaryCondition` instances).